crabs/detection_tracking/datasets.py

```diff
@@ -0,0 +1,190 @@
+                annotation_file = self.exclude_files(
+                    annotation_file, list_exclude_files
+                )
+
+            # create COCO dataset for detection
+            dataset_coco = CocoDetection(
+                img_dir,
+                annotation_file,
+                transforms=transforms,
+            )
+
+            # transform for "transforms v2"
+            dataset_transformed = wrap_dataset_for_transforms_v2(dataset_coco)
+            list_datasets.append(dataset_transformed)
+
+        # Concatenate datasets
+        full_dataset = torch.utils.data.ConcatDataset(list_datasets)
+
+        # Update class for this instance
+        self.__class__ = full_dataset.__class__
+        self.__dict__ = full_dataset.__dict__
+
+    def exclude_files(
+        self,
+        annotation_file: str,
+        list_files_to_exclude: list[str],
+    ) -> str:
+        """Remove selected images from annotation file and save new file.
+
+        A new annotation file is created, excluding the required images,
+        and without the annotations for those images.
+
+        Parameters
+        ----------
+        annotation_file : str
+            file with annotations
+        list_files_to_exclude : list[str]
+            list of filenames to exclude from the dataset
+
+        Returns
+        -------
+        str
+            path to new annotation file
+        """
+        # Read annotation file as a dataset dict
+        with open(annotation_file, "r") as f:
+            dataset = json.load(f)
+
+        # Determine images to exclude
+        slc_images_to_exclude = [
+            im["file_name"] in list_files_to_exclude
+            for im in dataset["images"]
+        ]
+        image_ids_to_exclude = [
+            im["id"]
+            for im, slc in zip(dataset["images"], slc_images_to_exclude)
+            if slc
+        ]
+
+        # Determine annotations to exclude
+        slc_annotations_to_exclude = [
+            ann["image_id"] in image_ids_to_exclude
+            for ann in dataset["annotations"]
+        ]
+
+        # Update dataset dict
+        dataset["images"] = [
+            im
+            for im, slc in zip(dataset["images"], slc_images_to_exclude)
+            if not slc
+        ]
+        dataset["annotations"] = [
+            annot
+            for annot, slc in zip(
+                dataset["annotations"], slc_annotations_to_exclude
+            )
+            if not slc
+        ]
+
+        # Write to new file under the same location as original annotation file
+        timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+        out_filename = Path(annotation_file).parent / Path(
+            Path(annotation_file).stem + "_filt_" + timestamp + ".json"
+        )
+        with open(out_filename, "w") as f:
+            json.dump(dataset, f)
+
+        return str(out_filename)
+
+
+def plot_sample(imgs: list, row_title: Optional[str] = None, **imshow_kwargs):
```

Comment:
yeah I mentioned this in the PR description (quote below). I found it very convenient to quickly sample and visualise the data but not sure here it's its best location.... Happy to remove it or move it to a visualisation module maybe?

> I got a helper plotting function from Pytorch vision examples, to easily plot samples from the data. But maybe this lives better somewhere else?
> ```
> sample = dataset[0]  # tuple with (image, annotations)
> plot_sample([sample])
> ```

Reply:
We agreed this function may come in handy and that it would be nice to have all viz utils together. To get started I moved the `plot_sample` convenience function to visualization module.